src/material/input/input.spec.ts

```diff
@@ -1240,7 +1243,7 @@ describe('MatInput with forms', () => {
       expect(containerEl.querySelectorAll('mat-error').length)
         .withContext('Expected no error messages when the input is valid.')
         .toBe(0);
-      expect(containerEl.querySelectorAll('mat-hint').length)
+      expect(Array.from(containerEl.querySelectorAll('mat-hint')).filter(isVisible).length)
```

Comment:
As mentioned above, I don't think that we should be using visibility to check the existence of the hints. This should use a selector similar to the one you added to the test harnesses.

Reply:
Updated!